torchgeo/datasets/geo.py

```diff
@@ -518,7 +465,10 @@ def __init__(
 
             self._res = res
 
-        self._crs = crs
+        # Create the dataset index
+        data = {'filepath': filepaths}
+        index = pd.IntervalIndex.from_tuples(datetimes, closed='both', name='datetime')
```

Comment:
I was thinking, right now we are creating a datetime index even though there are datasets which don't have any time information. This in combination with the min and max date of 1677 and 2262 becomes a bit confusing.  Maybe we can just check the length of `datetimes` and if it is empty we create a normal index instead of a IntervalIndex. Then for downstream steps, it is directly clear if the dataset has datetime information.

Reply:
Mixing index classes becomes complicated, because all of our getitem logic needs to support both types. The idea was that anything with no datetime info is static for all time. This is obviously not true in general, but it's a good assumption. But I'll keep this idea in mind.